Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/SimplePlayer.java

```diff
@@ -0,0 +1,109 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class SimplePlayer implements Player {
+
+	private boolean isNew=false;
+	private String name;
+	private Score mScore;
+	private final String DEFAULT_PATH="players.txt";
+
+	//--------------------------------------------
+	
+	public SimplePlayer (String nName){
+		if(!this.loadFromFile( DEFAULT_PATH,nName)){
+		isNew=true;
+		name = nName;
+		mScore = new SimpleScore();
+		}
+		
+	}
+	
+	public void writeToFile(){
+		writeToFile (DEFAULT_PATH);
+	}
+
+	public void writeToFile(String fileName){
+		try (FileWriter w = new FileWriter (fileName);){
+			w.write(name+"\n");
+			w.flush();
+			w.write(mScore.toString());
+			w.flush();
+		} catch (IOException e) {
+			e.printStackTrace();//Shouldn't happen
```

Comment:
Personally think its very reasonable for this to happen. The disk could fail. The folder might not exist. Could be read only. I think you should add a throws declaration to the method and let callers deal with it.